core/src/main/java/tc/oc/pgm/fireworks/FireworkMatchModule.java

```diff
@@ -0,0 +1,237 @@
+  private final boolean goalsEnabled = Config.Fireworks.goalsEnabled();
+  private final int number = 5;
+  private final int delay = 40;
+  private final int frequency = 40;
+  private final int iterationCount = 15;
+  private final int power = 2;
+
+  public FireworkMatchModule(Match match) {
+    this.match = match;
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onMatchEnd(final MatchFinishEvent event) {
+    if (!postEnabled) return;
+    match
+        .getScheduler(MatchScope.LOADED)
+        .runTaskTimer(delay, frequency, new FireworkRunner(event.getWinners()));
+  }
+
+  public static List<FireworkEffect.Type> AVAILABLE_TYPES =
+      ImmutableList.<FireworkEffect.Type>builder()
+          .add(Type.BALL)
+          .add(Type.BALL_LARGE)
+          .add(Type.BURST)
+          .add(Type.STAR)
+          .build();
+
+  public class FireworkRunner implements Runnable {
+    private final Set<Color> colors;
+    private final Collection<Competitor> winners;
+    private int iterations = 0;
+
+    public FireworkRunner(Collection<Competitor> winners) {
+      this.winners = winners;
+      this.colors =
+          winners.stream()
+              .map(winner -> BukkitUtils.colorOf(winner.getColor()))
+              .collect(Collectors.toSet());
+    }
+
+    @Override
+    public void run() {
+      if (this.iterations < iterationCount) {
+        // Build this list fresh every time, because MatchPlayers can unload, but Competitors can't.
+        final List<MatchPlayer> players =
+            winners.stream().flatMap(c -> c.getPlayers().stream()).collect(Collectors.toList());
+        Collections.shuffle(players);
+
+        for (int i = 0; i < players.size() && i < number; i++) {
+          MatchPlayer player = players.get(i);
+
+          Type type = AVAILABLE_TYPES.get(match.getRandom().nextInt(AVAILABLE_TYPES.size()));
+
+          FireworkEffect effect =
+              FireworkEffect.builder()
+                  .with(type)
+                  .withFlicker()
+                  .withColor(this.colors)
+                  .withFade(Color.BLACK)
+                  .build();
+
+          spawnFirework(player.getBukkit().getLocation(), effect, power);
+        }
+      }
+      this.iterations++;
+    }
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onWoolPlace(final PlayerWoolPlaceEvent event) {
+    if (goalsEnabled && event.getWool().isVisible()) {
+      this.spawnFireworkDisplay(
+          BlockVectors.center(event.getBlock()), event.getWool().getDyeColor().getColor(), 6, 2, 2);
+    }
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onCoreLeak(final CoreLeakEvent event) {
+    if (goalsEnabled && event.getCore().isVisible()) {
+      this.spawnFireworkDisplay(
+          event.getMatch().getWorld(),
+          event.getCore().getCasingRegion(),
+          event.getCore().getColor(),
+          8,
+          1.5,
+          2);
```

Comment:
What are these magic numbers? They should probably be static variables too.

Reply:
Also, why do we need static numbers? Can these variables not be derived by the size, location, etc.. of the objective?

Reply:
@Electroid To be honest I haven’t looked into why these numbers were set. As I stated in the original PR, most of this came right from the ProjectAres repo. So my guess is that these numbers are most likely finely tuned to whatever OCN used to have them as. 

When I performed testing of these changes everything worked really nicely, so I don’t really see a need to change them/risk having them look bad. If you want I can make these static variables as you suggested, but anything custom may take some time (I don’t have much free time for the next week).  

Let me know what you would like me to do. Thanks!

Reply:
Hmm, that's okay for now, we can skip over the magic numbers